python/pip_install/parse_requirements_to_bzl/extract_single_wheel/__init__.py

```diff
@@ -21,17 +21,31 @@ def main() -> None:
         required=True,
         help="A single PEP508 requirement specifier string.",
     )
+    parser.add_argument(
+        "--pip_platform_definition",
+        help="A pip platform definition in the form <platform>-<python_version>-<implementation>-<abi>",
+    )
     arguments.parse_common_args(parser)
     args = parser.parse_args()
     deserialized_args = dict(vars(args))
     arguments.deserialize_structured_args(deserialized_args)
 
     configure_reproducible_wheels()
 
-    pip_args = (
-        [sys.executable, "-m", "pip", "--isolated", "wheel", "--no-deps"] +
-        deserialized_args["extra_pip_args"]
-    )
+    pip_args = [sys.executable, "-m", "pip", "--isolated"]
+    if args.pip_platform_definition:
+        platform, python_version, implementation, abi = args.pip_platform_definition.split("-")
```

Comment:
Q: why this format instead of the format listed here in [PEP 425](https://www.python.org/dev/peps/pep-0425/#id8)? 

`{python tag}-{abi tag}-{platform tag}`

The extra bit in this PR's string is `{implementation}`, `cp = CPython`, but this is covered by the `{python tag}` from PEP 425. 

> The Python tag indicates the implementation and version required by a distribution.

Reply:
It's because `pip download` takes four flags: `--platform`, `--python-version`, `--implementation`, `--abi`. I'm not sure why pip doesn't line up with the PEP 425 format.